cli/compile/compile.go

```diff
@@ -127,6 +128,15 @@ func run(cmd *cobra.Command, args []string) {
 	}
 
 	sketchPath := initSketchPath(path)
+
+	// .pde files are still supported but deprecated, this warning urges the user to rename them
+	if files := sketches.CheckForPdeFiles(sketchPath); len(files) > 0 {
+		feedback.Error("Sketches with .pde extension are deprecated, please rename the following files to .ino:")
+		for _, f := range files {
+			feedback.Error(f)
+		}
+	}
+
```

Comment:
This piece is duplicated in compile.go, upload.go and archive.go.
On compile.go and upload.go there is also `initSketchPath()` that is duplicated as well.
This makes me wonder two things:
* in archive.go we are missing some cases because we miss an `initSketchPath(...)`
* we must move everything (the current initSketchPath and the pde check) into a common function `args.InitSketchParh(path)` or something like that...

Reply:
This is probably something that we need to tackle in a systematic way and not in this PR or it would make it explode.